src/combinations.rs

```diff
@@ -145,6 +145,16 @@ impl<I: Iterator> Combinations<I> {
     }
 }
 
+impl<I> Combinations<I>
+where
+    I: Iterator,
+    I::Item: Clone,
+{
+    fn current_combination(&self) -> Vec<I::Item> {
+        self.indices.iter().map(|i| self.pool[*i].clone()).collect()
```

Comment:
This is not what I meant but I just was not clear enough.
This code `indices.iter().map("index clone").collect()` do not appear only in `Combinations`, there is also `CombinationsWithReplacement`, `Permutations`, ~`MultiProduct`~ (oops, no I changed this one, good it simplifies things).
`LazyBuffer` could have a method `fn from_indices(&self, indices: &[usize]) -> Vec<I::Item>` that would be used in 3 iterators (and 3 files).

Reply:
Ah I see what you mean. Yes almost the exact same function is already implemented in `CombinationsWithReplacement`. I'll push a patch in a moment

Reply:
With `git grep "clone()).collect()"`, you would probably see most places to use this `from_indices` method.

Reply:
And you can remove this `current_combination` method, and probable similar method(s) in other iterators.